Sc 9946 split dockerfile into multiple stages

lukemarshhb opened the pull request:
- Added devcontainer.json for running inside vscode containers
- split Dockerfile into multiple stages
- replaced docker-compose with docker compose
- Updated the makefile
- Removed unused build scripts
- Removed all scripts for running outside docker
- Created base docker-compose file
- Make yarn default package manager

lukemarshhb committed:
use apk add --no-cache instead of add and update

--no-cache flag does the same as add and update

lukemarshhb committed:
replace bash with sh

bash isn't in alpine by default

lukemarshhb committed:
cleanup docker-compose.yml and extract common request_base

Simplify the file and make our environment variables consistent across services

lukemarshhb committed:
separate Dockerfile into stages

For more efficient builds. Also prevents dev dependencies from leaking into production.

lukemarshhb committed:
add a docker-compose dev profile

Gives us a way to run just the development services

lukemarshhb committed:
add production runtime to docker-compose

A way to run the services locally in production-like containers.

lukemarshhb committed:
Adding devcontainer so we can run in remote containers

lukemarshhb committed:
remove export node modules and clean up git hooks

lukemarshhb committed:
remove export COMPOSE_HTTP_TIMEOUT

lukemarshhb committed:
make yarn default package manager

lukemarshhb committed:
remove dev-all from Makefile

lukemarshhb committed:
fix test in makefile

marcpacheco-hubble commented on Dockerfile:
```diff
@@ -1,15 +1,30 @@
+FROM base AS deps
+# Install *only* the production dependencies
 COPY ["package.json", "yarn.*", "./"]
+RUN yarn
```
Do we need this line? or rather shouldn’t we run yarn after line 23 and delete this line?
possibly using `yarn install --production=true`

Because otherwise if you can’t install the dependancies you also can‘t get a shell and so you are stuck.

marcpacheco-hubble commented on Dockerfile:
```diff
@@ -1,15 +1,30 @@
-# Install app dependencies
-# with yarn
+FROM deps AS deps-dev
 RUN yarn
```
Shouldn’t we put this back re my previous comment

marcpacheco-hubble committed:
Use zsh as the container’s shell

Alpine’s default shell is ash, no human wants use that! So we have
added zsh for developer sanity.

marcpacheco-hubble committed:
New style Dockerfile/docker-compose.yml

We want to use multi stage dev/production builds so we have the right
code on the right places. With this approach we have smaller production
images and in dev we have all the things.
